backend/src/services/analytics/percentiles.py

```
@@ -0,0 +1,373 @@
+                PERCENTILE_CONT(0.90) WITHIN GROUP (ORDER BY {metric_name}) as p90,
+                PERCENTILE_CONT(0.95) WITHIN GROUP (ORDER BY {metric_name}) as p95,
+                PERCENTILE_CONT(0.99) WITHIN GROUP (ORDER BY {metric_name}) as p99,
+                AVG({metric_name}) as mean,
+                STDDEV({metric_name}) as std_dev,
+                MIN({metric_name}) as min,
+                MAX({metric_name}) as max,
+                COUNT(*) as count
+            FROM analytics.agent_runs
+            {where_clause}
+                AND {metric_name} IS NOT NULL
+        """)
```

Ruff (0.14.3)

Possible SQL injection vector through string-based query construction

(S608)